### ISSUE TITLE
[4.1] TinyMCE  - entering source code not possible

### ISSUE THREAD

**Issue by chmst** [joomla/joomla-cms#37294]
I got an E-mail from a blind user: 

[..] with the latest update to version 4.1.0 and the addition of the Source Code Plus option, I can no longer edit the source code of an article.
[..]
I'm not sure if this is Joomla itself or an update with the Tiny MCE Editor but, now that it is using an iFrame it is unusable.
Upon entering the iFrame, I seem to be presented with an editable field along with a read only list of the source code. Therefore I can not edit a line of code or insert HTML into the source code anymore.  
[..]

**Comment by brianteeman** [comment]
checking 

BTW its actually codemirror that is being opened in the iframe

**Comment by chmst** [comment]
Thank you. I was wondering as I cannot find an iframe for TinyMCE source code.

**Comment by brianteeman** [comment]
The code is located https://github.com/joomla/joomla-cms/tree/4.1-dev/build/media_source/plg_editors_tinymce/js/plugins/highlighter

And when built it lives in media/plg_editors_tinymce/js/plugins/highlighter

**Comment by dgrammatiko** [comment]
@chmst what where the steps they used?

I used `option + F10` to get to the toolbar and with `enter` on the code button the focus is moved in the iframe of the codemirror.

**Comment by chmst** [comment]
It is with screen reader. Unfortunately I have no clue how this works or how it worked before J4. I am very impressed by this letter and hope that we get much feedback from users. 

She wrote: 
[..]  Joomla has on the whole been accessible to myself using Apple's Voiceover component with the screen reader function.
I have taught myself a great deal using Joomla over the past 6 years and I am very grateful to the entire team for making this possible. I have constructed a number of sites using both Joomla 3.x and now 4.x but  

... continues as in my first post. 

Will try to get more info, and thanks for investigating

**Comment by dgrammatiko** [comment]
@chmst can you ask them to disable the code highlighting for the time?
I will try to test with voice over later today

**Comment by brianteeman** [comment]
It seemed ok to me on windows  with narrator but there are so many variables in play. browser, os, screenreader etc

**Comment by chmst** [comment]
I tried myself to swithc off the highlighting. In the tinyMCE plugin set the parameter "Source Code Highlighting" OFF. 
But this has no effect. The output highlighted.

**Comment by dgrammatiko** [comment]
>  The output highlighted.

That's a bug then

@chmst https://github.com/joomla/joomla-cms/pull/37298

**Comment by richard67** [comment]
Closing as having a pull request. Please test #37298 . Thanks in advance.

**Comment by dgrammatiko** [comment]
@richard67 the PR only solves the bug for the switch, the a11y part of this issue is still under investigation

**Comment by richard67** [comment]
@dgrammatiko Does your PR solve the issue completely? Or only partly?

**Comment by richard67** [comment]
Ah you were faster. Yes, I just noticed.

**Comment by chmst** [comment]
Thanks for the first part!

**Comment by brianteeman** [comment]
Is there any more feedback on what needs to be done?

**Comment by chmst** [comment]
Closing this as https://github.com/joomla/joomla-cms/pull/37298 solved the problem. 

Got a response right now: 

_All your hard work has paid off and the Source Code is now editable again using Apple Voiceover, it functions exactly as it used to. I can access the source code on an article and perform the required tasks.

This is wonderful news and I am so very grateful to yourself and all those who work so hard at keeping Joomla working so smoothly._

:)
